### PR TITLE
fix: deep_import on a repository loading 

### PR DESCRIPTION
**Pull request by howbazaar**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -12,10 +12,11 @@ jobs:
   Lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2.2.1
+      - uses: actions/checkout@v4
```
Comment (howbazaar) [inline .github/workflows/ci.yaml:15]: Drive by clean up of CI tasks. This will now honor the `.tool-versions`

File: .tool-versions [hunk 2]
```diff
@@ -1,3 +1,3 @@
 just 1.13.0
-poetry 1.5.1
-python 3.9.15
+poetry 1.8.3
```
Comment (howbazaar) [inline .tool-versions:2]: Now matches assetdb, platform, and executions.

File: tests/test_repository.py [hunk 3]
```diff
@@ -2,56 +2,50 @@
...
                         "url": "mock://git.acme.org/stacklet/policies.git",
                         "name": "test-policies",
                     }
                 }
-            },
+            )
+        return super().run_query("repository", args, response)
+
+    def test_add_repository(self):
+        res, body = self.run_query(
```
Comment (howbazaar) [inline tests/test_repository.py:29]: Hide most of the ickyness in a base method so the tests are cleaner.

**Review by johnsca — APPROVED** [review]
LGTM

**Review by johnsca — APPROVED** [review]

**Review by thisisshi — APPROVED** [review]

**Pull request merged**